src/services/errors/exception.filter.ts

```diff
@@ -1,22 +1,48 @@
 import { NextFunction, Request, Response } from 'express';
 import { IExceptionFilter } from './exception.filter.interface';
-import { HTTPError } from './http-error.class';
+import { Exception } from './exception.class';
 import { inject, injectable } from 'inversify';
 import { ILogger } from '../logger/logger.interface';
 import { TYPES } from '../../constants/constants';
 import 'reflect-metadata';
 import { StatusCodes } from 'http-status-codes';
+import { JsonWebTokenError, TokenExpiredError } from 'jsonwebtoken';
+import { Prisma } from '@prisma/client';
 
 @injectable()
 export class ExceptionFilter implements IExceptionFilter {
 	constructor(@inject(TYPES.ILogger) private logger: ILogger) {}
 
-	catch(err: Error | HTTPError, req: Request, res: Response, next: NextFunction): void {
-		if (err instanceof HTTPError) {
-			this.logger.error(`[${err.context}] Error ${err.statusCode}: ${err.message}`);
+	catch(
+		err: Error | Exception | JsonWebTokenError | TokenExpiredError,
+		req: Request,
+		res: Response,
+		next: NextFunction,
+	): void {
+		//console.log(err.constructor);
+		if (err instanceof Exception) {
+			this.logger.error(`[${err.context}] ${err.statusCode} ${err.name}: ${err.message}.`);
 			res.status(err.statusCode).send({ err: err.message });
+		} else if (err instanceof JsonWebTokenError || err instanceof TokenExpiredError) {
```

Comment:
We still need to proceed JsonWebTokenError because without covering it passed through to 500 internal server error. But tokens expiring errors proceed well without covering in exception filter.

TokenExpiredError removed